my-sms/spec/controllers/students_controller_spec.rb

```diff
@@ -0,0 +1,136 @@
+# is no simpler way to get a handle on the object needed for the example.
+# Message expectations are only used when there is no simpler way to specify
+# that an instance is receiving a specific message.
+#
+# Also compared to earlier versions of this generator, there are no longer any
+# expectations of assigns and templates rendered. These features have been
+# removed from Rails core in Rails 5, but can be added back in via the
+# `rails-controller-testing` gem.
+
+RSpec.describe StudentsController, type: :controller do
+  # This should return the minimal set of attributes required to create a valid
+  # Student. As you add validations to Student, be sure to
+  # adjust the attributes here as well.
+
+  # This should return the minimal set of values that should be in the session
+  # in order to pass any filters (e.g. authentication) defined in
+  # StudentsController. Be sure to keep this updated too.
+  let(:valid_session) { {} }
+
+  describe 'GET #index' do
+    it 'returns a success response' do
+      get :index, {}, valid_session
+      expect(response).to be_success
+    end
+  end
+
+  describe 'GET #show' do
+    it 'returns a success response' do
+      student = FactoryBot.create(:student)
+      get :show, { id: student.to_param }, valid_session
+      expect(response).to be_success
+    end
+  end
+
+  describe 'GET #new' do
+    it 'returns a success response' do
+      get :new, {}, valid_session
+      expect(response).to be_success
+    end
+  end
+
+  describe 'GET #edit' do
+    it 'returns a success response' do
+      student = FactoryBot.create(:student)
+      get :edit, { id: student.to_param }, valid_session
+      expect(response).to be_success
+    end
+  end
+
+  describe 'POST #create' do
+    context 'with valid params' do
+      it 'creates a new Student' do
+        expect do
+          post :create, { student: FactoryBot.attributes_for(:student) }, valid_session
+        end.to change(Student, :count).by(1)
+      end
+
+      it 'redirects to the created student' do
+        post :create, { student: FactoryBot.attributes_for(:student) }, valid_session
+        expect(response).to redirect_to(Student.last)
+      end
+    end
+
+    context 'with invalid params' do
+      it "returns a success response (i.e. to display the 'new' template)" do
+        invalid_student = FactoryBot.build(:student, first_name: nil)
+        post :create, { student: invalid_student }, valid_session
+        expect(response).to be_success
+      end
+    end
+  end
+
+  describe 'PUT #update' do
+    context 'with valid params' do
+      it 'updates the requested student' do
+        student = FactoryBot.create(:student)
+        new_attributes = FactoryBot.attributes_for(:student)
+        put :update, { id: student.to_param, student: new_attributes }, valid_session
+        student.reload
+
+        attrs = %i[first_name last_name gender]
+        student_attrs = student.attributes.fetch_values(*attrs.map(&:to_s))
```

Comment:
You can use the `%w` macro to create a list of strings rather than converting symbols to strings

Reply:
Had an issue where the new_attrs used symbols and the student_attrs wanted strings.